### ISSUE TITLE
sig-windows-gce test jobs are failing consistently for a long time

### ISSUE THREAD

**Issue by AnishShah** [kubernetes/kubernetes#124047]
### Which jobs are failing?

`gce-windows-2019-containerd-master` and `gce-windows-2022-containerd-master` are failing

### Which tests are failing?

The Windows nodes are failing to come up because the startup scripts are failing.

### Since when has it been failing?

It has been failing for a long time. It is red on the whole testgrid.

### Testgrid link

https://testgrid.k8s.io/sig-windows-gce

### Reason for failure (if possible)

The [Windows startup scripts](https://github.com/kubernetes/kubernetes/blob/master/cluster/gce/windows/k8s-node-setup.psm1) are failing to create NPD kubeconfig. I found this in the Windows Node serial log from a test run failure.

```
2024/03/25 15:15:33 GCEMetadataScripts: windows-startup-script-ps1: Exception caught in script:
2024/03/25 15:15:33 GCEMetadataScripts: windows-startup-script-ps1: At C:\k8s-node-setup.psm1:1549 char:14
2024/03/25 15:15:33 GCEMetadataScripts: windows-startup-script-ps1: +       -Token ${kube_env}['NODE_PROBLEM_DETECTOR_TOKEN']
2024/03/25 15:15:33 GCEMetadataScripts: windows-startup-script-ps1: +              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2024/03/25 15:15:33 GCEMetadataScripts: windows-startup-script-ps1: Kubernetes Windows node setup failed: Cannot bind argument to parameter 'Token' because it is an empty string.
2024/03/25 15:15:34 GCEMetadataScripts: windows-startup-script-ps1: Cleaning up, Unregistering WorkerServices...
2024/03/25 15:15:34 GCEMetadataScripts: windows-startup-script-ps1: [SC] OpenService FAILED 1060:
2024/03/25 15:15:34 GCEMetadataScripts: windows-startup-script-ps1:
2024/03/25 15:15:34 GCEMetadataScripts: windows-startup-script-ps1: The specified service does not exist as an installed service.
2024/03/25 15:15:34 GCEMetadataScripts: windows-startup-script-ps1:
2024/03/25 15:15:34 GCEMetadataScripts: windows-startup-script-ps1: [SC] OpenService FAILED 1060:
2024/03/25 15:15:34 GCEMetadataScripts: windows-startup-script-ps1:
2024/03/25 15:15:34 GCEMetadataScripts: windows-startup-script-ps1: The specified service does not exist as an installed service.
```

### Anything else we need to know?

On the Linux node, we run NPD in standalone mode and use the kubeconfig generated from the token specified in `NODE_PROBLEM_DETECTOR_TOKEN` or use the kubelet kubeconfig. [ref](https://github.com/kubernetes/kubernetes/blob/227c2e7c2b2c05a9c8b2885460e28e4da25cf558/cluster/gce/gci/configure-helper.sh#L3584-3594).

But on Windows, we do not run NPD because `ENABLE_NODE_PROBLEM_DETECTOR` is set to `none` based on the serial logs. 

```
2024/03/25 15:12:46 GCEMetadataScripts: windows-startup-script-ps1: Logging kube-env key-value pairs except CERT and KEY values
....
....
2024/03/25 15:12:46 GCEMetadataScripts: windows-startup-script-ps1: ENABLE_NODE_PROBLEM_DETECTOR: none
....
2024/03/25 15:12:46 GCEMetadataScripts: windows-startup-script-ps1: NODE_PROBLEM_DETECTOR_VERSION:
2024/03/25 15:12:46 GCEMetadataScripts: windows-startup-script-ps1: NODE_PROBLEM_DETECTOR_TOKEN:
```

Based on the Linux node behavior, We should maybe set `ENABLE_NODE_PROBLEM_DETECTOR` to standalone on Windows and use kubelet kubeconfig if `NODE_PROBLEM_DETECTOR_TOKEN` is missing?

### Relevant SIG(s)

/sig windows
/good-first-issue

**Comment by AryaMoghaddam** [comment]
I can give this a shot

**Comment by AnishShah** [comment]
/triage accepted
/assign @AryaMoghaddam 


Thanks! Reach out on slack if you need help

**Comment by kushalShukla-web** [comment]
i think i can work on this @AnishShah

**Comment by lavishpal** [comment]
@AryaMoghaddam are you still working on this issue?

**Comment by AryaMoghaddam** [comment]
@lavishpal I switched to another issue, sorry for the delay

**Comment by Khagesh16** [comment]
@AnishShah Can I work on this issue? I had a look at it earlier, I think I need to use gce on google cloud to reproduce. Is it possible to reproduce on local? Please, let me know if there any docs. Thanks

**Comment by shwetd19** [comment]
Hey there!

I'm Shwetas Dhake, known as @shwetd19 on GitHub. I've been using Docker and Kubernetes lately and really enjoy what I'm learning. I saw issue #124047 and thought, why not jump in and see if I can help out? 

I'm not super experienced yet, but I'm eager to learn and get stuck in. If you think I could be helpful for this issue, I'd love to give it a shot. 

Thanks for considering

**Comment by SequoiaGod** [comment]
Hi mate, 
If you wanna give it a try, just type "/assign" like what I did before. Currently, This issue is blocked by sorts of testing issues, but yeah give it a try mate.

**Comment by Warlord77** [comment]
@AnishShah  I can also work on this
